hadoop-pipeline-application/src/main/java/org/molgenis/hadoop/pipeline/application/cachedigestion/ContigRegionsMap.java

```diff
@@ -0,0 +1,255 @@
+package org.molgenis.hadoop.pipeline.application.cachedigestion;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Ordering;
+
+/**
+ * {@link Map} that for each contig contains a {@code key} representing the contig name combined with as {@code value}
+ * an {@link ImmutableList} with all {@link Region}{@code s} having that contig name. Uses an {@link ImmutableList} to
+ * ensure that the {@link Region}{@code s} are sorted and indexable.
+ */
+public class ContigRegionsMap implements Map<String, ImmutableList<Region>>
+{
+	/**
+	 * The {@link Region} {@link ImmutableList}{@code s} (value), stored per contig (key) in a {@link Map}.
+	 */
+	private Map<String, ImmutableList<Region>> contigRegions = new HashMap<>();
+
+	/**
+	 * Number of {@link Region}{@code s} stored.
+	 */
+	private int nRegions = 0;
+
+	@Override
+	public int size()
+	{
+		return contigRegions.size();
+	}
+
+	/**
+	 * Returns the total number of {@link Region}{@code s} stored.
+	 * 
+	 * @return {@code int}
+	 */
+	public int numberOfRegions()
+	{
+		return nRegions;
```

Comment:
It would be less error prone to just calculate this value on the fly, instead of letting multiple methods bookkeep this value. Unless you've actually observed this to be a performance bottleneck.